apps/mark-scan/backend/src/server.test.ts

```diff
@@ -1,25 +1,102 @@
-import { fakeLogger } from '@votingworks/logging';
+import { LogEventId, fakeLogger } from '@votingworks/logging';
 import tmp from 'tmp';
 import { buildMockInsertedSmartCardAuth } from '@votingworks/auth';
+import {
+  BooleanEnvironmentVariableName,
+  getFeatureFlagMock,
+} from '@votingworks/utils';
+import { MockPaperHandlerDriver } from '@votingworks/custom-paper-handler';
+// import { sleep } from '@votingworks/basics';
 import { PORT } from './globals';
-import { start } from './server';
+import { resolveDriver, start } from './server';
 import { createWorkspace } from './util/workspace';
-import { getMockStateMachine } from '../test/app_helpers';
+
+const featureFlagMock = getFeatureFlagMock();
+jest.mock('@votingworks/utils', (): typeof import('@votingworks/utils') => {
+  return {
+    ...jest.requireActual('@votingworks/utils'),
+    isFeatureFlagEnabled: (flag: BooleanEnvironmentVariableName) =>
+      featureFlagMock.isEnabled(flag),
+  };
+});
+
+beforeEach(() => {});
+
+afterEach(() => {
+  jest.resetAllMocks();
+});
 
 test('can start server', async () => {
   const auth = buildMockInsertedSmartCardAuth();
   const logger = fakeLogger();
   const workspace = createWorkspace(tmp.dirSync().name);
-  const stateMachine = await getMockStateMachine(workspace, logger);
 
   const server = await start({
     auth,
     logger,
     port: PORT,
     workspace,
-    stateMachine,
   });
   expect(server.listening).toBeTruthy();
   server.close();
-  stateMachine.stopMachineService();
+  workspace.reset();
 });
+
+test('resolveDriver returns a mock driver if feature flag is on', async () => {
+  featureFlagMock.enableFeatureFlag(
+    BooleanEnvironmentVariableName.SKIP_PAPER_HANDLER_HARDWARE_CHECK
+  );
+  const logger = fakeLogger();
+
+  const driver = await resolveDriver(logger);
+  expect(driver).toBeInstanceOf(MockPaperHandlerDriver);
+  expect(logger.log).toHaveBeenCalledWith(
+    LogEventId.PaperHandlerConnection,
+    'system',
+    {
+      disposition: 'failure',
+    }
+  );
+});
+
+// Something about the feature flag mock (?) makes these tests not terminate cleanly
```

Comment:
looks like it's not uncommented. @kshen0 I'm not sure what it would have to do with the feature flag mock, but you could reset all flags between tests (it doesn't happen automatically, but maybe we could set that up)

Reply:
[7024b8e](https://github.com/votingworks/vxsuite/pull/4628/commits/7024b8e922b2c2e0eccd8b2266f8e1e5628c55af)

I did a little digging and it might be more related to the business logic of the flag than the flag mocking. I was able to add back in the mocked auth test but the mocked driver test had the same problem, so I removed it entirely and cleaned up the comments.